### PR TITLE
[v2.9] Migrating to the AWS SDK for Go V2

### PR DESCRIPTION
**Pull request by mjura**
The AWS SDK v1 (which is used by this project) is moving to maintenance mode July 2024 and will be completely out of support July 2025.

With the long deprecation policy of Rancher we will still have the eks-operator around when the AWS SDK v1 moves out of support.
We need to update the operator and migrate to v2 of the SDK.

Issue: https://github.com/rancher/eks-operator/issues/336 (cherry picked from commit 6ec4368bb4a76c5b16120d7b1f83c74632f80d8c)

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**
Issue #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits into logical changes
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests
- [ ] backport needed

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -5,7 +5,14 @@ go 1.21
 replace k8s.io/client-go => k8s.io/client-go v0.28.6
 
 require (
-	github.com/aws/aws-sdk-go v1.49.24
+	github.com/aws/aws-sdk-go v1.50.34
```
Comment (yiannistri) [inline go.mod:8]: Is the previous version still used somewhere?

File: .github/workflows/lint.yaml [hunk 2]
```diff
@@ -20,3 +20,4 @@ jobs:
         uses: golangci/golangci-lint-action@v3
         with:
           args: -v
+          skip-pkg-cache: true
```
Comment (yiannistri) [inline .github/workflows/lint.yaml:23]: Shall we add a comment here to describe why skipping the pkg cache is helpful?

**Review by yiannistri — DISMISSED** [review]
Nice work :+1: My suggestions are mostly to aid comprehension/readability but this PR looks good.

File: controller/internal.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	})
+	if doesNotExist(err) {
+		name = oldStyleName
+	}
+
+	_, err = svc.DeleteStack(ctx, &cloudformation.DeleteStackInput{
+		StackName: aws.String(name),
+	})
+	if err != nil && !doesNotExist(err) {
+		return fmt.Errorf("error deleting stack: %v", err)
```
[the remark this answers is not in the document]
Comment (mjura) [reply controller/internal.go]: fixed

File: go.mod (hunk 1 above)
Comment (mjura) [reply go.mod:8]: it will be addressed in another PR

**Review by yiannistri — APPROVED** [review]

**Pull request merged**